ui/src/common/components/__tests__/CitationSummaryGraph.test.jsx

```diff
@@ -1,79 +1,340 @@
+        from: 0,
+        to: 1,
+        doc_count: 10,
+      };
+      const data = wrapper.instance().toSeriesData(bucket, PUBLISHED_BAR_TYPE);
+      const expectedData = {
+        x: '0--0',
+        y: 10,
+        label: '10',
+        color: ORANGE,
+      };
+      expect(data).toEqual(expectedData);
+    });
+    it('returns series data with correct color for hovered bar', () => {
+      const wrapper = shallow(
+        <CitationSummaryGraph
+          publishedData={mockPublishedData}
+          citeableData={mockCiteableData}
+          loadingCitationSummary={false}
+          error={null}
+          onSelectBarChange={jest.fn()}
+        />
+      );
+      wrapper.setState({
+        hoveredBar: { type: PUBLISHED_BAR_TYPE, xValue: '0--0' },
+      });
+      const bucket = {
+        key: '0--0',
+        from: 0,
+        to: 1,
+        doc_count: 10,
+      };
+      const data = wrapper.instance().toSeriesData(bucket, PUBLISHED_BAR_TYPE);
+      const expectedData = {
+        x: '0--0',
+        y: 10,
+        label: '10',
+        color: HOVERED_ORANGE,
+      };
+      expect(data).toEqual(expectedData);
+    });
+    it('returns series data with correct color when selected bar', () => {
+      const wrapper = shallow(
+        <CitationSummaryGraph
+          publishedData={mockPublishedData}
+          citeableData={mockCiteableData}
+          loadingCitationSummary={false}
+          error={null}
+          onSelectBarChange={jest.fn()}
+          selectedBar={{ type: CITEABLE_BAR_TYPE, xValue: '0--0' }}
+        />
+      );
+      const bucketSelectedBar = {
+        key: '0--0',
+        from: 0,
+        to: 1,
+        doc_count: 10,
+      };
+      const dataSelectedBar = wrapper
+        .instance()
+        .toSeriesData(bucketSelectedBar, CITEABLE_BAR_TYPE);
+      const expectedDataSelectedBar = {
+        x: '0--0',
+        y: 10,
+        label: '10',
+        color: BLUE,
+      };
+      expect(dataSelectedBar).toEqual(expectedDataSelectedBar);
+
+      const bucketUnSelectedBar = {
+        key: '10--49',
+        from: 0,
+        to: 1,
+        doc_count: 10,
+      };
+      const dataUnSelectedBar = wrapper
+        .instance()
+        .toSeriesData(bucketUnSelectedBar, CITEABLE_BAR_TYPE);
+      const expectedDataUnSelectedBar = {
+        x: '10--49',
+        y: 10,
+        label: '10',
+        color: GRAY,
+      };
+      expect(dataUnSelectedBar).toEqual(expectedDataUnSelectedBar);
+    });
+  });
+  describe('onCiteableBarClick', () => {
+    it('calls onSelectBarChange with correct bar', () => {
+      const onSelectBarChange = jest.fn();
+      const wrapper = shallow(
+        <CitationSummaryGraph
+          publishedData={mockPublishedData}
+          citeableData={mockCiteableData}
+          loadingCitationSummary={false}
+          error={null}
+          onSelectBarChange={onSelectBarChange}
+        />
+      );
+      wrapper.instance().onCiteableBarClick({ x: '0--0' });
```

Comment:
I know we have tests that are like this but it would be better (more real) if you  do
```
it('calls onSelectedBarChange when citable bar clicked on the grap')
  ...
  const onCiteableBarClick = wrapper.find([data-test-id=citeable-bar-series]).prop('onValueClick');
  onCiteableBarClick(...)
   ...
  expect(onSelectBarChange).toHaveBeenCalledWith...
```